### PR TITLE
Master configurators owl pocalypse vfe

### PR DESCRIPTION
**Pull request by Feyensv**
See also:
- https://github.com/odoo/enterprise/pull/31522

### CODE REVIEW

File: addons/web/static/src/views/basic_relational_model.js [hunk 1]
```diff
@@ -1034,12 +1034,15 @@ export class StaticList extends DataPoint {
...
-        this.model.__bm__.notifyChanges(parentID, {
+        await this.model.__bm__.notifyChanges(parentID, {
             order_line: {
                 operation: "MULTI",
                 commands: commandsWithId,
             },
         });
+
+        this.model.root.__syncData();
+        this.model.notify();
```
Comment (Feyensv) [inline addons/web/static/src/views/basic_relational_model.js:1045]: c'est le js qui a proposé de modifier tout ça? :eyes:

Comment (hote-odoo) [reply addons/web/static/src/views/basic_relational_model.js:1045]: yep, I added the function following their instructions,  the last two lines `__syncData` and `notify` were recently added and are needed to apply changes to the rest of the lines and re-render the changes

File: addons/sale/__manifest__.py [hunk 2]
```diff
@@ -67,11 +67,10 @@
             'sale/static/src/scss/product_configurator.scss',
             'sale/static/src/js/sale_progressbar_field.js',
             'sale/static/src/js/tours/sale.js',
-            'sale/static/src/js/sale_order_line_mixin.js',
-            'sale/static/src/js/product_configurator_widget.js',
-            'sale/static/src/js/sale_order_controller.js',
-            'sale/static/src/js/sale_order_view.js',
-            'sale/static/src/js/product_discount_widget.js',
+            'sale/static/src/js/product_configurator_widget_legacy.js',
```
Comment (Feyensv) [inline addons/sale/__manifest__.py:70]: to remove

File: addons/web/static/src/views/basic_relational_model.js [hunk 3]
```diff
@@ -1021,6 +1021,30 @@ export class StaticList extends DataPoint {
...
+            return {
+                operation: command.operation,
+                id: command.record.__bm_handle__,
+                data: command.data,
+            };
+        });
+
+        const parentID = this.model.__bm__.localData[this.__bm_handle__].parentID;
+        await this.model.__bm__.notifyChanges(parentID, {
+            order_line: {
```
Comment (Feyensv) [inline addons/web/static/src/views/basic_relational_model.js:1038]: forbidden to have sale ref in web :D

File: addons/web/static/src/views/basic_relational_model.js [hunk 4]
```diff
@@ -1021,6 +1022,31 @@ export class StaticList extends DataPoint {
         await this.__syncParent(operation);
     }
 
+    /**
+     * @param {Array[]} commands  array of commands
+     */
+    async applyBatchedCommands(commands) {
```
Comment (FrancoisGe) [inline addons/web/static/src/views/basic_relational_model.js:1028]:
```suggestion
    async applyCommands(commands) {
```

**Comment by Feyensv** [comment]
Moved to #101304

**Pull request closed without merging**